### PR TITLE
Feature/component button

### PR DESCRIPTION
**Pull request by Urszuja**
Reusable button with styles and animation on hover (koala) - with common onClick (log with source button)

### CODE REVIEW

File: features/buttonComponent.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/* 
+Component accepts following inputs: label (string), class (string), animate on hover (boolean)
+Component has a light grey background and black text. When hovered, the background changes to dark grey and the label is white.
+If animate on hover is true, the button will show an icon of Koala to the right of it
+*/
+import './buttonComponentStyles.css';
+
+export default function createButton(label, classStyle, animate) {
+  const koalaButton = document.createElement('div');
+  koalaButton.classList.add('koala-button');
```
Comment (Nefariusek) [inline features/buttonComponent.js:10]: Try to use this part only if animate === true - we don;t want every button to be wrapped in koala stuff

**Review by Nefariusek — CHANGES_REQUESTED** [review]
Move code to the proper folder. Avoid creating new directories with code outside src. 

try this approach:

src
&nbsp;|- components
&nbsp;&nbsp;&nbsp; |- Button (hold all source files here. You can omit Styles and Component in file names) 

  Any Image file keep in the 'public' folder

File: src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+};
+
+export default function Button(label, className, animate, eventListener, callback) {
+  const button = document.createElement('button');
+  button.innerHTML = label;
+  button.classList.add(className);
+
+  button.addEventListener(eventListener, callback);
+
+  animate && addKoala();
```
Comment (Nefariusek) [inline src/components/Button.js:29]: I'm not a big fan of this approach 
I find more readable
`
if(animate) {
  addKoala(): 
}
`

File: src/components/Button.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    koala.classList.add('hidden');
+  };
+};
+
+export default function Button(label, className, animate, eventListener, callback) {
+  const button = document.createElement('button');
+  button.innerHTML = label;
+  button.classList.add(className);
+
+  button.addEventListener(eventListener, callback);
```
Comment (Nefariusek) [inline src/components/Button.js:27]: I wonder if we would ever need this "eventListener" to be not "click"

**Review by Nefariusek — APPROVED** [review]
I posted additional comments (just tips). It can be merged

**Comment by Nefariusek** [comment]
@Urszuja, ready to merge!

File: src/components/Button.js (hunk 3 above)
Comment (Urszuja) [reply src/components/Button.js:27]: just in case 🐼

**Pull request merged**